pkg/ocm/fakes/fakes.go

```diff
@@ -10,13 +11,19 @@ import (
 	ocmctrl "github.com/open-component-model/ocm-controller/pkg/ocm"
 )
 
+// getResourceReturnValues defines the return values of the GetResource function.
+type getResourceReturnValues struct {
+	reader io.ReadCloser
+	err    error
+}
+
 // MockFetcher mocks OCM client. Sadly, no generated code can be used, because none of them understand
 // not importing type aliased names that OCM uses. Meaning, external types request internally aliased
 // resources and the mock does not compile.
 // I.e.: counterfeiter: https://github.com/maxbrunsfeld/counterfeiter/issues/174
 type MockFetcher struct {
-	getResourceReader                   io.ReadCloser
-	getResourceErr                      error
+	getResourceCallCount                int
+	getResourceReturns                  map[int]getResourceReturnValues
```

Comment:
I only extended this function because there was no need for the rest. Whoever needs a CallN function can extend the mock if need be.